Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1150,6 +1151,136 @@ Reading 2.390625MB ...
 
 To learn more, please refer to the [r2frida wiki](https://github.com/enovella/r2frida-wiki/blob/master/README.md "r2frida Wiki").
 
+##### Information Gathering
+
+In previous section we learned about using radare2 to collect runtime information of a running process. Radare2 is internally using Frida (r2frida) to obtain process information. In this section we will learn how to use Frida directly to gather process related information.
+
+##### Getting Loaded Classes and their Methods
+
+In Frida REPL Objective-C runtime related information can be accessed using `ObjC` command. Within `ObjC` command, `enumerateLoadedClasses` function lists the loaded classes for a given application.
+
+```
+$ frida -U -f com.iOweAss
+
+[iPhone::com.iOweAss]-> ObjC.enumerateLoadedClasses()
```

Comment:
Which app is this, iOweAss?

Reply:
It is a crackme app I [wrote](https://github.com/su-vikas/crackmes/tree/master/ios), and was present on my device while writing this. The uncrackable level 1 dont have enough classes to show this example. Is it fine?

Reply:
Got it. Can you please remove the "Ass" :-)

Reply:
Is this app publicly available?

Reply:
haha sure. Yes, I linked above, it is here:  https://github.com/su-vikas/crackmes

Reply:
Fixed!